nav2_simple_commander/nav2_simple_commander/footprint_collision_checker.py

```diff
@@ -0,0 +1,96 @@
+#! /usr/bin/env python3
+# Copyright 2021 Samsung Research America
+# Copyright 2022 Afif Swaidan
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from math import cos, sin
+from line_iterator import LineIterator
+from costmap_2d import PyCostmap2D
+from geometry_msgs.msg import Point
+
+NO_INFORMATION = 255
+LETHAL_OBSTACLE = 254
+INSCRIBED_INFLATED_OBSTACLE = 253
+MAX_NON_OBSTACLE = 252
+FREE_SPACE = 0
+
+
+class FootprintCollisionChecker():
+    def __init__(self, costmap: PyCostmap2D):
+        self.costmap_ = costmap
```

Comment:
Remove the costmap setting on constructor and only on `setCostmap`. The fact that the C++ version does both is an oddity due to history that we can start fresh from in Python

Reply:
I agree with you, I found it was redundant but I was taking the C++ version as a reference, I prefer to remove it as well.
Will do that in the next commit tomorrow hopefully.